### ISSUE TITLE
Consider changing return type of `Cancelable.empty` from `Empty` back to `Cancelable`

### ISSUE THREAD

**Issue by sirmax** [monix/monix#773]
3.0.0-RC1 had `Cancelable.empty` return `Cancelable`, 3.0.0-RC2 changed the return type to `Empty`.

This change breaks compilation of fold-like code, for example this compiles with 3.0.0-RC1
```
Observable
  .intervalAtFixedRate(…)
  .scan(Cancelable.empty -> …) {
    case ((prevKillSwitch, _), i) =>
      prevKillSwitch.cancel()
      newKillSwitch = BooleanCancelable()
      newKillSwitch -> …
  }
```
but with 3.0.0-RC2 scalac complains that the expression in `scan` does not conform to expected type `(Cancelable.Empty, …)`.

Using `Cancelable()` instead of `Cancelable.empty` makes it compile again, but it would be nice to have `empty` also work since it's a bit more expressive.

If there are any valid reasons to keep `empty` as it is in RC2, feel free to close this issue :)

**Comment by alexandru** [comment]
@sirmax will consider reverting it.

In the meantime you can also do: `Cancelable.empty : Cancelable`.
This is a type-checked upcast.

```scala
source.scan(Cancelable.empty : Cancelable) { ??? }
```

**Comment by sirmax** [comment]
`Cancelable.empty : Cancelable` is too wordy to my taste, I'll keep using `Cancelable()` for now which is effectively `Cancelable.empty`.

**Comment by alexandru** [comment]
Well, `Cancelable.empty` is empty, meaning it doesn't allocate anything, it's the same instance being reused over and over again.

**Comment by alexandru** [comment]
I think you're right btw, this isn't a very nice change.

**Comment by sirmax** [comment]
> Well, `Cancelable.empty` is empty, meaning it doesn't allocate anything

But `Cancelable()` is the same
```
object Cancelable {
  /** Builds a [[monix.execution.Cancelable Cancelable]]. */
  def apply(): Cancelable =
    empty
```
Although yes, I know, one shouldn't rely on an implementation :)
